application/src/main/java/org/togetherjava/tjbot/commands/help/BookmarksSystem.java

```diff
@@ -0,0 +1,185 @@
+     *
+     * @param event The command interaction event
+     */
+    void requestRemovePagination(GenericCommandInteractionEvent event) {
+        Objects.requireNonNull(removePaginationConsumer,
+                "No remove pagination consumer was provided");
+
+        removePaginationConsumer.accept(event);
+    }
+
+    boolean isHelpThread(MessageChannelUnion channel) {
+        if (channel.getType() != ChannelType.GUILD_PUBLIC_THREAD) {
+            return false;
+        }
+
+        ThreadChannel threadChannel = channel.asThreadChannel();
+        String parentChannelName = threadChannel.getParentMessageChannel().getName();
+
+        return isOverviewChannelName.test(parentChannelName);
+    }
+
+    boolean didUserBookmarkChannel(long userID, long channelID) {
+        return database.read(context -> context.selectFrom(BOOKMARKS)
+            .where(BOOKMARKS.AUTHOR_ID.eq(userID), BOOKMARKS.CHANNEL_ID.eq(channelID))
+            .fetch()
+            .isNotEmpty());
+    }
+
+    void addBookmark(long authorID, long channelID, @Nullable String note) {
+        database.write(context -> context.newRecord(BOOKMARKS)
+            .setAuthorId(authorID)
+            .setChannelId(channelID)
+            .setCreatedAt(Instant.now())
+            .setNote(note)
+            .setScheduledRemovalAt(null)
+            .insert());
+    }
+
+    List<BookmarksRecord> getUsersBookmarks(long authorID) {
+        return database.read(context -> context.selectFrom(BOOKMARKS)
+            .where(BOOKMARKS.AUTHOR_ID.eq(authorID))
+            .orderBy(BOOKMARKS.CREATED_AT.desc())
+            .fetch());
+    }
+
+    void removeBookmarks(long authorID, List<Long> channelIDs) {
+        database.write(context -> context.deleteFrom(BOOKMARKS)
+            .where(BOOKMARKS.AUTHOR_ID.eq(authorID), BOOKMARKS.CHANNEL_ID.in(channelIDs))
+            .execute());
+    }
+
+    int getTotalBookmarkCount() {
+        return database
+            .read(context -> context.selectCount().from(BOOKMARKS).fetchOne(0, int.class));
+    }
+
+    int getUserBookmarkCount(long authorID) {
+        return database.read(context -> context.selectCount()
+            .from(BOOKMARKS)
+            .where(BOOKMARKS.AUTHOR_ID.eq(authorID))
+            .fetchOne(0, int.class));
+    }
+
+    void scheduleUsersBookmarksRemoval(long authorID) {
+        Instant scheduledRemovalAt = Instant.now().plus(LEAVE_BOOKMARKS_REMOVAL_DELAY);
+
+        database.write(context -> context.update(BOOKMARKS)
+            .set(BOOKMARKS.SCHEDULED_REMOVAL_AT, scheduledRemovalAt)
+            .where(BOOKMARKS.AUTHOR_ID.eq(authorID))
+            .execute());
+    }
+
+    void cancelUsersBookmarksRemoval(long authorID) {
+        database.write(context -> context.update(BOOKMARKS)
+            .setNull(BOOKMARKS.SCHEDULED_REMOVAL_AT)
+            .where(BOOKMARKS.AUTHOR_ID.eq(authorID))
+            .execute());
+    }
+
+    void cleanRemovalScheduledBookmarks() {
```

Comment:
odd name. maybe `deleteExpiredBookmarks()`

Reply:
Hm. I myself am unsure on how to name the removal system. They dont really expire.

Reply:
but u consider it expired after that time 🤷 or maybe "stale" or "irrelevant". in either case, try to find an adjective that describes the property. rather than calling it "bookmark that is scheduled for deletion" and then coming up with names like "cleanup bookmarks scheduled for deletion" rather than "delete expired/stale/irrelevant bookmarks"

Reply:
How about `deleteLeftoverBookmarks`?

Reply:
also good